fix luatex for libpoppler 0.12
patch by Ilya Barygin
licensed under GPLv2, see <20091022102009.208b5935@corner> on
debian-tex-maint  Debian mailing list
---
 source/texk/web2c/luatexdir/image/pdftoepdf.cc |   18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

Index: luatex-0.60.0/source/texk/web2c/luatexdir/image/pdftoepdf.cc
===================================================================
--- luatex-0.60.0.orig/source/texk/web2c/luatexdir/image/pdftoepdf.cc	2010-04-12 23:35:17.000000000 +0900
+++ luatex-0.60.0/source/texk/web2c/luatexdir/image/pdftoepdf.cc	2010-04-12 23:35:31.000000000 +0900
@@ -471,7 +471,8 @@
     int rotate;
     char *checksum;
     PDFRectangle *pagebox;
-    float pdf_version_found, pdf_version_wanted, xsize, ysize, xorig, yorig;
+    unsigned major_pdf_version_found, minor_pdf_version_found;
+    float xsize, ysize, xorig, yorig;
     assert(idict != NULL);
     assert(img_type(idict) == IMG_TYPE_PDF);
     assert(readtype == IMG_CLOSEINBETWEEN);     // only this is implemented
@@ -494,15 +495,18 @@
     // this works only for PDF 1.x -- but since any versions of PDF newer
     // than 1.x will not be backwards compatible to PDF 1.x, pdfTeX will
     // then have to changed drastically anyway.
-    pdf_version_found = pdf_doc->doc->getPDFVersion();
-    pdf_version_wanted = 1 + (minor_pdf_version_wanted * 0.1);
-    if (pdf_version_found > pdf_version_wanted + 0.01) {
+    major_pdf_version_found = pdf_doc->doc->getPDFMajorVersion();
+    minor_pdf_version_found = pdf_doc->doc->getPDFMinorVersion();
+    if (major_pdf_version_found > 1 ||
+        (major_pdf_version_found == 1 &&
+         minor_pdf_version_found > minor_pdf_version_wanted)
+       ) {
         char msg[] =
-            "PDF inclusion: found PDF version <%.1f>, but at most version <%.1f> allowed";
+            "PDF inclusion: found PDF version <%u.%u>, but at most version <1.%i> allowed";
         if (pdf_inclusion_errorlevel > 0) {
-            pdftex_fail(msg, pdf_version_found, pdf_version_wanted);
+            pdftex_fail(msg, major_pdf_version_found, minor_pdf_version_found, minor_pdf_version_wanted);
         } else {
-            pdftex_warn(msg, pdf_version_found, pdf_version_wanted);
+            pdftex_warn(msg, major_pdf_version_found, minor_pdf_version_found, minor_pdf_version_wanted);
         }
     }
     img_totalpages(idict) = pdf_doc->doc->getCatalog()->getNumPages();
